### PR TITLE
Refactor MQTT-SN code

### PR DESCRIPTION
**Pull request by embhorn**
Move MQTT-SN code into new files to better organize codebase

### CODE REVIEW

File: examples/sn-client/sn-client_qos-1.c [hunk 1]
```diff
@@ -28,7 +28,7 @@
     #include <config.h>
 #endif
 
-#include "wolfmqtt/mqtt_client.h"
+#include "wolfmqtt/mqtt_sn_client.h"
```
Comment (dgarske) [inline examples/sn-client/sn-client_qos-1.c:31]: For backwards compatibility you could have `mqtt_client.h` conditionally include `wolfmqtt/mqtt_sn_client.h` automatically based on `WOLFMQTT_SN`.

**Review by dgarske — CHANGES_REQUESTED** [review]
Wonderful!! Thank you

File: src/mqtt_sn_client.c [hunk 2]
```diff
@@ -0,0 +1,1872 @@
+/* sn_mqtt_client.c
```
Comment (dgarske) [inline src/mqtt_sn_client.c:1]: `sn_mqtt_client.c` -> `mqtt_sn_client.c`

File: src/mqtt_sn_packet.c [hunk 3]
```diff
@@ -0,0 +1,1576 @@
+/* sn_mqtt_packet.c
```
Comment (dgarske) [inline src/mqtt_sn_packet.c:1]: `sn_mqtt_packet.c` -> `mqtt_sn_packet.c`.

File: src/mqtt_sn_client.c [hunk 4]
```diff
@@ -0,0 +1,1872 @@
...
+            break;
+        }
+    } /* switch (packet_type) */
+
+    (void)packet_id;
+
+    return rc;
+}
+
+static int SN_Client_WaitType(MqttClient *client, void* packet_obj,
```
Comment (dgarske) [inline src/mqtt_sn_client.c:457]: As a future TODO... when comparing this with `MqttClient_WaitType` there are some fixes that need brought over.

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]

File: src/mqtt_sn_client.c (hunk 4 above)
Comment (embhorn) [reply src/mqtt_sn_client.c:457]: Created #367

**Pull request merged**